File: ChefApp/app/src/main/java/com/emedinaa/chefapp/data/DataInjector.java
package com.emedinaa.chefapp.data;

import android.content.Context;
import androidx.annotation.NonNull;

import com.emedinaa.chefapp.BuildConfig;
import com.emedinaa.chefapp.data.remote.APIClient;
import com.emedinaa.chefapp.data.socket.SocketManager;

/**
 * @author : Eduardo Medina
 * @see : https://developer.android.com/index.html
 * @since : 8/4/18
 */
public class DataInjector {

    private static DataInjector instance;

    private APIClient remote;
    private @NonNull
    Context context;
    //private Socket mSocket;
    private SocketManager socketManager;

    private DataInjector(){}

    public static synchronized DataInjector getInstance(){
        if(instance == null){
            instance = new DataInjector();
        }
        return instance;
    }

    public void setUp(@NonNull Context mContext){
        context= mContext;
        remote=new APIClient(BuildConfig.DOMAIN);
        socketManager= new SocketManager();
        socketManager.init();
        /*try {
            mSocket = IO.socket(BuildConfig.DOMAIN);
        } catch (URISyntaxException e) {
            throw new RuntimeException(e);
        }*/
    }

    public APIClient remote(){
        return this.remote;
    }

    /*public Socket socket(){
        return mSocket;
    }*/

    public SocketManager socketManager() {
        return socketManager;
    }
}

File: RestaurantApp/app/src/main/java/com/emedinaa/kotlinapp/data/DataInjector.kt
package com.emedinaa.kotlinapp.data

import android.content.Context
import com.emedinaa.kotlinapp.dish.DishesRepository
import com.emedinaa.kotlinapp.home.CategoriesRepository
import com.emedinaa.kotlinapp.login.LogInRepository
import com.emedinaa.kotlinapp.socket.SocketManager

object DataInjector {

    private lateinit var apiClient:ApiClient.ServicesApiInterface
    private lateinit var logInRepository:LogInRepository
    private lateinit var dishesRepository: DishesRepository
    private lateinit var categoriesRepository: CategoriesRepository
    private lateinit var socketManager:SocketManager

    fun setUp(context:Context){
        apiClient= ApiClient().build()
        socketManager= SocketManager()

        logInRepository = LogInRepository(apiClient)
        dishesRepository = DishesRepository(apiClient)
        categoriesRepository = CategoriesRepository(apiClient)
    }

    fun provideLogInRepository():LogInRepository{
        return logInRepository
    }

    fun provideDishesRepository():DishesRepository{
        return dishesRepository
    }


    fun provideCategoriesRepository():CategoriesRepository{
        return categoriesRepository
    }

    fun provideSocketManager():SocketManager{
        return socketManager
    }
}